### ISSUE TITLE
If state doesn't exist, it should raise an error

### ISSUE THREAD

**Issue by asmaier** [dr-prodigy/python-holidays#597]
This is confusing:

```
>>> import date
>>> import holidays
>>> date(2021,3,8) in holidays.CountryHoliday("DE", state="BE")    # should raise an error, because state="BE" does not exist
False
>>> date(2021,3,8) in holidays.CountryHoliday("DE", prov="BE")
True
```
The first line after the imports should better raise an error, because for Germany one has to use `prov` and not `state`. At the moment this leads to subtle bugs in computing wether a day is a holiday or not.

**Comment by mborsetti** [comment]
@dr-prodigy I have a proposed solution.

According to the [README](https://github.com/dr-prodigy/python-holidays/blob/master/README.rst), we don't have a single country where we use **both** `state `**and** `prov`.  Since this is the case, I propose we simplify everything by conforming to [ISO 3166-2](https://en.wikipedia.org/wiki/ISO_3166-2#Current_codes) and call these **subdivisions** (argument name `subdivision` or `subd`) and make everything simple.

We can maintain backwards compatibility by assigning the value of either `state` or `prov`, if specified, to `subdivision`.

If you agree I can do global changes this weekend and submit a PR.

**Comment by dr-prodigy** [comment]
Hey @asmaier
thank you for pointing this out. This inconsistency in fact has already been raised and would in fact require a review.

@mborsetti
thank you, I like your solution a lot! If you wish to implement it, I'll absolutely bless it :-)
Then - I believe - we could/should possibly review the current subdivision names to adhere to the reported ISO codes, but that can come at a later time.

Thanks both! 👍

**Comment by mborsetti** [comment]
@dr-prodigy 
> I like your solution a lot! If you wish to implement it, I'll absolutely bless it :-)

Cool.  

Will do the **subdivisions**, but first I need to make sure that #586 and #602 are merged/commented into beta branch because of the interdependencies with the documentation and I don't want to have to deal with the hell of trying to merge (or rebase?) commits back into my branch. I almost lost it today with #600, #601, and #602 (I think I need to take a class on `git`!)

What name do you prefer for the parameter: `subdivision=`,  `subdiv=` or `subd=`?

> Then - I believe - we could/should possibly review the current subdivision names to adhere to the reported ISO codes, but that can come at a later time.

Yup, later time.

**Comment by dr-prodigy** [comment]
Hey there @mborsetti !
Regarding your question, I didn't have a chosen answer, to be honest :-)
Given that we already have another shortened property (`prov` instead of province) I would then go for `subdiv`.

Regarding the other 2 PRs, I've just merged #602 , while I kindly ask you to review #586 : I've added some notes on it.

No worries about git, it's more about you always do complex / extensive tasks, and this expose you to more difficulties.. but you're doing very fine! :-) 
Thank you very much, best
